app/actions/websocket/threads.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {handleThreadArrived, handleReadChanged, handleAllMarkedRead, handleFollowChanged, getThread as fetchThread} from '@mm-redux/actions/threads';
+import {getCurrentUserId} from '@mm-redux/selectors/entities/common';
+import {getCurrentTeamId} from '@mm-redux/selectors/entities/teams';
+import {getThread} from '@mm-redux/selectors/entities/threads';
+import {ActionResult, DispatchFunc, GetStateFunc} from '@mm-redux/types/actions';
+import {WebSocketMessage} from '@mm-redux/types/websocket';
+
+export function handleThreadUpdated(msg: WebSocketMessage) {
+    return (dispatch: DispatchFunc, getState: GetStateFunc): ActionResult => {
+        try {
+            const threadData = JSON.parse(msg.data.thread);
+            handleThreadArrived(dispatch, getState, threadData, msg.broadcast.team_id);
+        } catch {
+            // invalid JSON
```

Comment:
I know the try/catch here is to avoid crashes if the json is invalid, but why would be the json be invalid, can you elaborate on the comment